storage/src/main/java/org/apache/uniffle/storage/handler/impl/PooledHdfsShuffleWriteHandler.java

```diff
@@ -55,7 +55,7 @@ public PooledHdfsShuffleWriteHandler(
     // todo: support init lazily
     try {
       for (int i = 0; i < maxConcurrency; i++) {
-        queue.offer(
+        queue.add(
```

Comment:
We are sure the capacity will not be exceeded, use `add()` instead.
Note: `add()` throws `IllegalStateException` when queue is full.